tests/toolchain/build-pass/separate_cpp_hpp.hpp

```diff
@@ -1,6 +1,6 @@
 #include <eosio/eosio.hpp>
 
-class [[eosio::contract]] separate_cpp_hpp : eosio::contract {
+class [[eosio::contract]] separate_cpp_hpp : public eosio::contract {
```

Comment:
Is this requirement a breaking change?

Reply:
I wonder how common that is. I see in the system contracts we `: public contract`.

It might be easy enough to solve this by changing for example this
https://github.com/AntelopeIO/cdt/blob/a294a08ab742f4c10977da668125a091a211344d/tools/include/eosio/codegen.hpp#L304-L306
to
```cpp
static_cast<eosio::contract&>(obj).set_exec_type(eosio::contract::exec_type_t::call)
```
afaict that's safe, but I'm not really sure if we need to bother or not